From 974224a1cd2a3c9a16806346769cf0d6b8e657ee Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 27 Jan 2015 17:26:48 +0800
Subject: [PATCH 0091/5242] MLK-10101-5 usb: chipidea: otg: init otg_fsm of
 usb_bus while start host

commit  818f2e4bf86a360b70300e3fe4f89b8e809f24cc from
https://source.codeaurora.org/external/imx/linux-imx.git

Set otg_fsm of usb_bus to be ci->fsm while start host.

Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit 65efc6c5f3936b254a77cf02478d1268585def84)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/host.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/chipidea/host.c b/drivers/usb/chipidea/host.c
index c2ebe3d..876a022 100644
--- a/drivers/usb/chipidea/host.c
+++ b/drivers/usb/chipidea/host.c
@@ -300,6 +300,7 @@ static int host_start(struct ci_hdrc *ci)
 		ci->hcd = hcd;
 
 		if (ci_otg_is_fsm_mode(ci)) {
+			hcd->self.otg_fsm = &ci->fsm;
 			otg->host = &hcd->self;
 			hcd->self.otg_port = 1;
 		}
-- 
1.7.9.5

